### PR TITLE
Treat "expired certs" errors returned from RPCs as retryable on the client side

### PR DESCRIPTION
**Pull request by gzdunek**
Since Connect will no longer check the user cert before making an RPC, it has to be able to properly recognize "expired certs" errors that come from the server (to show a re-login dialog). 
To make it work, I added an exception for the two errors `IsErrorResolvableWithRelogin`. Ideally, we should just return a "returnable" error directly from the client as described by Alan, but this sounds like a big task, beyond the scope of caching clients in Connect.

Depends on https://github.com/gravitational/teleport/pull/38202.

### CODE REVIEW

File: lib/auth/auth_with_roles.go [hunk 1]
```diff
@@ -2888,7 +2888,7 @@ func (a *ServerWithRoles) generateUserCerts(ctx context.Context, req proto.UserC
 			return nil, trace.AccessDenied("access denied")
 		}
 		if req.Expires.Before(a.authServer.GetClock().Now()) {
-			return nil, trace.AccessDenied("access denied: client credentials have expired, please relogin.")
+			return nil, utils.ErrClientCredentialsHaveExpired
```
Comment (codingllama) [inline lib/auth/auth_with_roles.go:2891]:
```suggestion
			return nil, trace.Wrap(utils.ErrClientCredentialsHaveExpired)
```

File: lib/utils/utils.go [hunk 2]
```diff
@@ -682,6 +682,11 @@ func ByteCount(b int64) string {
 // ErrLimitReached means that the read limit is reached.
 var ErrLimitReached = &trace.LimitExceededError{Message: "the read limit is reached"}
 
+// ErrClientCredentialsHaveExpired means
+// that the credentials expired on the server-side,
+// and the user should relogin.
+var ErrClientCredentialsHaveExpired = &trace.AccessDeniedError{Message: "access denied: client credentials have expired, please relogin."}
```
Comment (codingllama) [inline lib/utils/utils.go:688]: Could we find a more "thematic" place for this? Maybe api/client/errors.go, unless someone has a better idea?

File: lib/utils/utils.go [hunk 3]
```diff
@@ -682,6 +682,11 @@ func ByteCount(b int64) string {
 // ErrLimitReached means that the read limit is reached.
 var ErrLimitReached = &trace.LimitExceededError{Message: "the read limit is reached"}
 
+// ErrClientCredentialsHaveExpired means
+// that the credentials expired on the server-side,
+// and the user should relogin.
```
Comment (codingllama) [inline lib/utils/utils.go:687]: nit: Use longer lines?

```suggestion
// ErrClientCredentialsHaveExpired means that the credentials expired on the
// server-side, and the user should relogin.
```

**Review by codingllama — APPROVED** [review]
I appreciate you using the "guard" error instead of just matching strings for both, that is already a bit better.

**Comment by codingllama** [comment]
(Also I approved a bit quickly, please address the comments.)

File: lib/utils/utils.go (hunk 2 above)
Comment (gzdunek) [reply lib/utils/utils.go:688]: I found the `var ErrLimitReached` error which sits above, and there I added mine.
But that error is used only locally, in `utils.go` 🤦.

`api/client/errors.go` sounds good to me.

**Review by codingllama — APPROVED** [review]

**Review by ravicious — APPROVED** [review]
I feel like this change can be merged alone, without waiting for the parent PR. The change in `lib/teleterm/apiserver/middleware.go` could be accounted for in #38202 itself.

**Comment by gzdunek** [comment]
Good point, @ravicious. I will merge this PR to master, the git history will be cleaner.

**Pull request merged**